pkg/entities/ie.go

```diff
@@ -399,7 +401,19 @@ func EncodeToIEDataType(dataType IEDataType, val interface{}, buff *bytes.Buffer
 			err := util.Encode(buff, binary.BigEndian, uint32(ipv4Int.Uint64()))
 			return uint32(ipv4Int.Uint64()), err
 		} else {
-			return v, nil
+			return 0, fmt.Errorf("provided IP does not belong to IPv4 address family")
```

Comment:
Do we need to change the way of decoding IPv4 similar to IPv6: directly encode the result from `To4()` method?

Reply:
That is an extra check if someone gives the wrong IPv6 address with incorrect length. I created separate case blocks for easy code readability.

Reply:
Got that. I mean for line 399-401, maybe we can change to `err := util.Encode(buff, binary.BigEndian, v)` as well. (similar to the way you did for IPv6 to make code more tidy)

Reply:
Yes that can be done. I was under the impression `To4()` may return IPv4 address with a length of 16bytes. Looking at the function closely, it only returns 4 bytes. Thanks for pointing. Updated the patch.